### PR TITLE
Web-components: Add full reload listening to server-side-events

### PR DESCRIPTION
**Pull request by brion-fuller**
Issue: #12307

Web components cannot be hot reloaded as customElement.define can only be called once per page with an unique string. This has been on the users to add to their config file. Now this is available for free.

## What I did
I hooked into server-sent events for webpack HMR and listen for the built event in which I trigger a page reload.

## How to test
Is this testable with Jest or Chromatic screenshots? no
Does this need a new example in the kitchen sink apps? no
Does this need an update to the documentation? yes

### CODE REVIEW

File: examples/web-components-kitchen-sink/.storybook/main.js [hunk 1]
```diff
@@ -12,6 +13,6 @@ module.exports = {
     '@storybook/addon-viewport',
   ],
   core: {
-    builder: 'webpack4',
+    builder: 'webpack5',
```
Comment (shilman) [inline examples/web-components-kitchen-sink/.storybook/main.js:16]: AFAIK we can't use this in the monorepo

**Review by shilman — CHANGES_REQUESTED** [review]

File: examples/web-components-kitchen-sink/.storybook/main.js (hunk 1 above)
Comment (shilman) [inline examples/web-components-kitchen-sink/.storybook/main.js:16]:
```suggestion
    builder: 'webpack4',
```

**Review by shilman — APPROVED** [review]
LGTM, thanks @brion-fuller!! @gaetanmaisse the CI results looks like an intermittent NPM failure so I'm merging anyway.

**Comment by steveblue** [comment]
Thank you @brion-fuller confirmed this works with vanilla custom elements.

**Pull request merged**